stacks-common/src/util/hash.rs

```diff
@@ -382,12 +382,15 @@ pub struct MerklePathPoint<H: MerkleHashFunc> {
 pub type MerklePath<H> = Vec<MerklePathPoint<H>>;
 
 /// Merkle tree implementation with tagged nodes:
-/// * a leaf hash is H(0x00 + data)
-/// * a node hash is H(0x01 + left.hash + right.hash)
-/// An empty tree has root hash 0x00000...00000
+///
+/// * A leaf hash is `H(0x00 + data)`
+/// * A node hash is `H(0x01 + left.hash + right.hash)`
+///
+/// An empty tree has a root hash of `0x00000...00000`.
 ///
 /// NOTE: This is consensus-critical code, because it is used to generate the transaction Merkle
 /// tree roots in Stacks blocks.
+#[allow(clippy::ptr_arg)]
```

Comment:
I guess this is because of:
```
pub fn new(data: &Vec<Vec<u8>>) -> MerkleTree<H> {
```

Any problem with just changing that to use a ` &[Vec<u8>]` instead of adding the `allow`?

Reply:
I have no problem doing it! was just worried it would be too invasive but will do!